src/main/java/uk/gov/hmcts/reform/logging/layout/ReformLoggingLayout.java

```diff
@@ -37,6 +42,26 @@ public void setRequireAlertLevel(boolean requireAlertLevel) {
         this.requireAlertLevel = requireAlertLevel;
     }
 
+    public void setRequireErrorCode(boolean requireErrorCode) {
+        this.requireErrorCode = requireErrorCode;
+    }
+
+    private void appendExtraExceptionFlags(StringBuilder log, ILoggingEvent event) {
+        AbstractLoggingException exception = null;
+
+        if (requireAlertLevel || requireErrorCode) {
+            exception = AbstractLoggingException.getFromLogEvent(event);
+        }
+
+        if (exception != null && requireAlertLevel) {
```

Comment:
It would be perfect to do a similar type of optimisation here as you did in the other place - putting these two conditions in the body of the first one.